multiscale_run/bloodflow_manager.py

```diff
@@ -72,15 +77,25 @@ def get_seg_points(self, scale):
 
         return self.seg_points * scale
 
-    def get_flows(self):
+    def get_flows(self, idxs: list = None):
         """Get the flow in each vasculature segment.
 
         This method retrieves the blood flow values associated with each vasculature segment in the loaded vasculature graph.
 
-        Returns
-            numpy.ndarray: An array containing the flow values for each vasculature segment.
+        Args:
+            idxs (list, optional): A list of indices to select specific segments from the flow array.
+                                    If not provided, the full array is returned.
+
+        Returns:
+            numpy.ndarray: An array containing the flow values for each vasculature segment,
+            or a selection based on the provided indices.
         """
-        return self.graph.edge_properties["flow"].to_numpy()
+        flow_array = self.graph.edge_properties["flow"].to_numpy()
+
+        # If idxs are provided, return the values at those idxs, otherwise return the full array
+        if idxs is not None:
+            return flow_array[idxs]
+        return flow_array
```

Comment:
If `idxs` is not `None`, do you know if it is possible to call `self.graph.edge_properties["flow"][idxs]` to prevent allocating a NumPy array for nothing?

Reply:
I can check that

Reply:
it was not as simple as that because pandas.Series does not allow you to do that kind of indexing but I have found a workaround